### DISCUSSION TITLE
What's the best/most performant way to build a giant mega menu in Remix?

### DISCUSSION THREAD

**Discussion by dangayle** [remix-run/remix#8647] [Q&A]
My company is doing a PoC for Remix and while we're wrapping our heads around route configurations, I'm tasked with figuring out the best way to handle a large mega menu with dropdowns, flyouts, all the normal busy stuff.

What's the best way to handle loading and building of that data without incurring giant performance hits on page navigations?

**Comment by lili21** [comment]
you better use nested route as many you can, can split the data loading to each route, and you will be fine.
